pkg/ottl/ottlfuncs/func_replace_all_matches.go

```diff
@@ -69,6 +71,17 @@ func replaceAllMatches[K any](target ottl.PMapGetter[K], pattern string, replace
 				return nil, fmt.Errorf("replacement value is not a string")
 			}
 			replacementVal = replacementValStr
+			if !replacementFormat.IsEmpty() { // If replacementFormat is not empty, add it to the replacement value
```

Comment:
Since the validation here is not trivial, could we extract this into a function shared by all four OTTL functions?

Reply:
I've moved this validation to a shared function.